### ISSUE TITLE
InfoNCE of pre-defined anchor, positive and negative pairs

### ISSUE THREAD

**Issue by alaaj27** [KevinMusgrave/pytorch-metric-learning#686]
To begin with, I appreciate the tremendous amount of work that has been put into this repo! I have a question about using [InfoNCE](https://arxiv.org/pdf/1807.03748.pdf) loss. 

I have a pre-defined anchor (Nxd) , one positive (Nxd), and multiple negatives (NxMxd), where N is the number of triplets in the batch, d is the vector embedding size, and M is the number of negatives per one anchor. The question now is how can we calculate the InfoNCE loss of anchor, positive, and negative?

**Comment by KevinMusgrave** [comment]
I think this should work:
```python
from pytorch_metric_learning.losses import NTXentLoss

loss_fn = NTXentLoss()

a = torch.arange(0, N).repeat(M)
p = torch.arange(N, N*2).repeat(M)
n = torch.arange(N*2, N*2 + N*M)

loss = loss_fn(embeddings, indices_tuple=(a,p,n))
```
